tools/Generator/Generator.cs

```diff
@@ -728,7 +732,14 @@ struct FixedParamInfo
 					bool isFixed = fixedParams != null &&  fixedParams.ContainsKey (name);
 					string paramName = isFixed ? "ptr" + name : name;
 
-					Write ("{0}{1}{2}", (isPointer && !isStruct && !isFixed) ? "&" : "", keywords.Contains (paramName) ? "@" + paramName : paramName, (!isFixed && (isStruct || isHandle)) ? ".m" : "");
+					if (isOptionalParam && isPointer && !isOut)
+					{
+						Write ("{0} != null ? {0}{1} : null", keywords.Contains (paramName) ? "@" + paramName : paramName, (!isFixed && (isStruct || isHandle)) ? ".m" : "");
```

Comment:
Please split out the expression `!isFixed && (isStruct || isHandle)` into a separate variable and use the variable here and below.